nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/XmlFlowEncryptor.java

```diff
@@ -0,0 +1,120 @@
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Characters;
+import javax.xml.stream.events.StartDocument;
+import javax.xml.stream.events.XMLEvent;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class XmlFlowEncryptor implements FlowEncryptor {
+    private static final Pattern ENCRYPTED_PATTERN = Pattern.compile("enc\\{([^\\}]+?)\\}");
+
+    private static final int FIRST_GROUP = 1;
+
+    private static final String ENCRYPTED_FORMAT = "enc{%s}";
+
+    private static final XMLEventReaderProvider eventReaderProvider = new StandardXMLEventReaderProvider();
+
+    private final Logger logger = LoggerFactory.getLogger(XmlFlowEncryptor.class);
+
+    @Override
+    public void processFlow(InputStream inputStream, OutputStream outputStream, PropertyEncryptor inputEncryptor, PropertyEncryptor outputEncryptor) {
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        XMLOutputFactory xmlOutputFactory = XMLOutputFactory.newInstance();
+        final XMLEventFactory eventFactory = XMLEventFactory.newInstance();
+        try {
+            final XMLEventReader reader = eventReaderProvider.getEventReader(new StreamSource(inputStream));
+            final XMLEventWriter writer = xmlOutputFactory.createXMLEventWriter(outputStream, "UTF-8");
+            while (reader.hasNext()) {
+                final XMLEvent event = reader.nextEvent();
+                if (event.getEventType() == XMLEvent.START_DOCUMENT) {
+                    final StartDocument startDocument = (StartDocument) event;
+                    handleXmlStartDocument(startDocument, outputStream);
+                } else if (event.getEventType() == XMLEvent.CHARACTERS) {
+                    final Characters characters = event.asCharacters();
+                    final String value = characters.getData();
+                    final Matcher matcher = ENCRYPTED_PATTERN.matcher(value);
+                    if (matcher.matches()) {
+                        final String processedValue = getOutputEncrypted(matcher.group(FIRST_GROUP), inputEncryptor, outputEncryptor);
+                        writer.add(eventFactory.createCharacters(processedValue));
+                    } else {
+                        writer.add(characters);
+                    }
+                } else if (event.getEventType() == XMLEvent.COMMENT) {
+                    writer.add(event);
+                    final Characters newLine = eventFactory.createCharacters("\n");
+                    writer.add(newLine);
```

Comment:
Is there a particular reason for adding the newline? It seems best to avoid adding the newline character.